pennylane/tape/tape.py

```diff
@@ -1555,8 +1582,18 @@ def copy(self, copy_operations=False):
             tape._ops = self._ops.copy()
             tape._measurements = self._measurements.copy()
 
-        tape._update()
+        tape._graph = None
+        tape._specs = None
+        tape._depth = None
+        tape.wires = copy.copy(self.wires)
+        tape.num_wires = self.num_wires
+        tape.is_sampled = self.is_sampled
+        tape.all_sampled = self.all_sampled
+        tape._update_par_info()
```

Comment:
```suggestion
        tape._par_info = copy.copy(self._par_info)
```
Might as well.  Keeps the pattern.

Reply:
As `_par_info` contains operations, I was not sure whether a shallow copy is enough here? :thinking:

Reply:
Deep copy then?
```suggestion
        tape._par_info = copy.deepcopy(self._par_info)
```

Reply:
I checked, and neither `copy.deepcopy` nor `copy.copy` works with the currently desired behaviour (as per the tests).
When using `deepcopy`, the "totally shallow" (i.e. including operations) copying behaviour breaks (`test_tape.TestTapeCopying.test_shallow_copy`), when using `copy`, the shallow copying that copies operations separately breaks (`test_tape.TestTapeCopying.test_shallow_copy_with_operations`)

I'd leave it to be `_update_par_info` for now, or is there an alternative still without writing a separate function? :thinking:

Reply:
Not that important of a thing.  Let's just leave it be with `tape._update_par_info()`.